samza-test/src/test/java/org/apache/samza/test/framework/SamzaApplicationFailureTests.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.test.framework;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.TimeoutException;
+import org.apache.samza.SamzaException;
+import org.apache.samza.application.StreamApplication;
+import org.apache.samza.operators.KV;
+import org.apache.samza.system.OutgoingMessageEnvelope;
+import org.apache.samza.system.SystemStream;
+import org.apache.samza.test.controlmessages.TestData;
+import org.apache.samza.test.framework.stream.CollectionStream;
+import org.junit.Test;
+
+
+public class SamzaApplicationFailureTests {
```

Comment:
There're existing integration tests for test-runner api viz `StreamApplicationIntegrationTest` and `StreamTaskIntegrationTest` .  Since the new tests are testing error scenarios for streamApplication and streamTask, it will be better to move them to the existing test classes.

Reply:
`StreamApplicationIntegrationTest` and  `StreamTaskIntegrationTest` will be serving as example to users, hence I wanted to keep tests for samza seprately(based on category). We might also move them (StreamApplicationIntegrationTest, StreamTaskIntegrationTest and AsyncStreamTaskIntegrationTest) to src for examples (if needed)

Thanks

Reply:
Ideally `test-runner` api samples should be part of https://github.com/apache/samza-hello-samza (which already houses samples for high-level, low-level API using `standalone` and `yarn` or samza-samples in main codebase).

IMHO, It should not be placed in samza-test module(contains integration tests for components) and shared with users.

Having duplicate test classes responsible for same thing will create confusion. 

It'll be better to combine these test classes.